beacon-chain/p2p/subnets.go

```diff
@@ -198,30 +231,14 @@ func (s *Service) findPeersWithSubnets(
 			continue
 		}
 
-		// Remove duplicates, keeping the node with higher seq.
-		existing, ok := nodeByNodeID[node.ID()]
-		if ok && existing.Seq() > node.Seq() {
-			continue
-		}
-		nodeByNodeID[node.ID()] = node
-
 		// We found a new peer. Modify the defective subnets map
 		// and the filter accordingly.
-		for subnet := range defectiveSubnets {
-			if !nodeSubnets[subnet] {
-				continue
-			}
-
-			defectiveSubnets[subnet]--
-
-			if defectiveSubnets[subnet] == 0 {
-				delete(defectiveSubnets, subnet)
-			}
+		nodeByNodeID[node.ID()] = node
 
-			filter, err = s.nodeFilter(topicFormat, defectiveSubnets)
-			if err != nil {
-				return nil, errors.Wrap(err, "node filter")
-			}
+		updateDefectiveSubnets(nodeSubnets, defectiveSubnets)
+		filter, err = s.nodeFilter(topicFormat, defectiveSubnets)
```

Comment:
I moved this filter call outside of the loop because I don't see how defective subnets would update in the loop. Probably an error keeping it in